addons/web/static/src/views/calendar/utils.js

```diff
@@ -37,15 +37,16 @@ export function getColor(key) {
     return colorMap.get(key);
 }
 
-export function getFormattedDateSpan(start, end) {
+export function getFormattedDateSpan(start, end, options = {}) {
     const isSameDay = start.hasSame(end, "days");
 
     if (!isSameDay && start.hasSame(end, "month")) {
         // Simplify date-range if an event occurs into the same month (eg. "August 4-5, 2019")
         return start.toFormat("LLLL d") + "-" + end.toFormat("d, y");
     } else {
+        const format = options.sameMonthFormat ?? "DDD";
```

Comment:
option name doesn't seem correct, as you're specifically in the case where start and end aren't in the same month. The given format will be used either if
 - start and end are the same day
 - start and end are from different months

I'm not sure it makes sense to allow to specify a custom format that would only be used in those cases, but not if start and end are in the same month (different days). Depending on the provided format, it could lead to inconsistencies, which is the case with the one you provided actually (display of the day in some cases, but not always). Do you *really* need this?